ws/portforward/model.go

```diff
@@ -15,14 +15,29 @@
 package portforward
 
 const (
-	MessageTypePortForwardNew  = "new"
+	// MessageTypePortForwardNew is the message type to start a port-forwarding connection
+	// The body MUST contain a PortForwardNew object.
+	MessageTypePortForwardNew = "new"
+	// MessageTypePortForwardStop is the message type to stop a port-forwarding connection
 	MessageTypePortForwardStop = "stop"
-	MessageTypePortForward     = "forward"
+	// MessageTypePortForward is the message type for streaming data
+	MessageTypePortForward = "forward"
+	// MessageTypeError is returned on internal or protocol errors. The
+	// body MUST contain an Error object.
+	MessageTypePortError = "error"
 )
 
-const (
-	PropertyPortForwardConnectionID = "connection_id"
-	PropertyPortForwardRemoteHost   = "remote_host"
-	PropertyPortForwardRemotePort   = "remote_port"
-	PropertyPortForwardProtocol     = "protocol"
-)
+// The Error struct is passed in the Body of MsgProto in case the message type is ErrorMessage
+type PortForwardNew struct {
+	RemoteHost *string `msgpack:"remote_host" json:"remote_host"`
+	RemotePort *uint   `msgpack:"remote_port" json:"remote_port"`
+	Protocol   *string `msgpack:"protocol" json:"protocol"`
```

Comment:
This is `tcp`/`udp` right?

Reply:
Yes

Reply:
I reworked it as a custom type and provided the possible values as constants.